crates/wasmi/src/store.rs

```diff
@@ -236,6 +268,17 @@ impl StoreInner {
             elems: Arena::new(),
             extern_objects: Arena::new(),
             fuel: Fuel::default(),
+
+            // Counts and limits of instances, memories and tables tracked by
+            // ResourceLimiter. Counts are kept separate from the sizes of arenas
+            // above in order to allow checking "up front" in a single early and
+            // fallible call to [Store::bump_resource_counts].
```

Comment:
Cannot this check be performed uppon bumping as well with a check like

```rust
if store.memories.len() == store.memory_limit {
    return Ok(false);
}
```
?
It would be great if we could get rid of this redundant information. Always bug prone.

Reply:
I think not "up front" / early when this method is called -- the memories haven't been allocated yet. I was trying to follow the structure of wasmtime in terms of how and when things get checked. I can try harder to get rid of this though.

Reply:
low priority issue though. highest priorities are getting CI happy and fixing those benchmarks. everything else can be done later or in a follow-up.